homeassistant/helpers/entity_component.py

```diff
@@ -270,9 +270,15 @@ async def _async_reset(self) -> None:
 
     async def async_remove_entity(self, entity_id: str) -> None:
         """Remove an entity managed by one of the platforms."""
+        found = None
+
         for platform in self._platforms.values():
             if entity_id in platform.entities:
-                await platform.async_remove_entity(entity_id)
+                found = platform
+                break
+
+        if found:
+            await found.async_remove_entity(entity_id)
```

Comment:
Why not just add a `break`?

```suggestion
                await platform.async_remove_entity(entity_id)
                break
```

Reply:
We're using an AST tool to look for code that is using await inside loops. In this case that's not a problem since we can break after the first await but often we want to avoid awaiting in loops since it's inefficient.

By rewriting the code we avoid a false positive when using the AST tool in the future.

Reply:
Thanks